src/gps/src/gpsServiceClientNode.cpp

```diff
@@ -0,0 +1,257 @@
+                );
+                return false;
+            }
+
+            if (future_result.get()->success) {
+                RCLCPP_INFO(
+                    get_logger(), "Transition %d successfully triggered.", static_cast<int>(transition)
+                );
+                return true;
+            } else {
+                RCLCPP_WARN(
+                    get_logger(), "Failed to trigger transition %u", static_cast<unsigned int>(transition)
+                );
+                return false;
+            }
+        }
+
+    private:
+        std::shared_ptr<rclcpp::Client<lifecycle_msgs::srv::GetState>> client_get_state_;
+        std::shared_ptr<rclcpp::Client<lifecycle_msgs::srv::ChangeState>> client_change_state_;
+};
+
+
+void callee_script(std::shared_ptr<LifecycleServiceClient> gps_client){
+    rclcpp::WallRate time_between_state_changes(0.1);  // 10s
+
+    // configure
+    {
+        if (!gps_client->change_state(lifecycle_msgs::msg::Transition::TRANSITION_CONFIGURE)) {
+            return;
+        }
+        if (!gps_client->get_state()) {
+            return;
+        }
+    }
+
+    // activate
+    {
+        time_between_state_changes.sleep();
+        if (!rclcpp::ok()) {
+            return;
+        }
+        if (!gps_client->change_state(lifecycle_msgs::msg::Transition::TRANSITION_ACTIVATE)) {
+            return;
+        }
+        if (!gps_client->get_state()) {
+            return;
+        }
+    }
+
+    // deactivate
+    {
+        time_between_state_changes.sleep();
+        if (!rclcpp::ok()) {
+            return;
+        }
+        if (!gps_client->change_state(lifecycle_msgs::msg::Transition::TRANSITION_DEACTIVATE)) {
+            return;
+        }
+        if (!gps_client->get_state()) {
+            return;
+        }
+    }
+
+    // activate
+    {
+        time_between_state_changes.sleep();
+        if (!rclcpp::ok()) {
+            return;
+        }
+        if (!gps_client->change_state(lifecycle_msgs::msg::Transition::TRANSITION_ACTIVATE)) {
+            return;
+        }
+        if (!gps_client->get_state()) {
+            return;
+        }
+    }
+
+    // deactivate
+    {
+        time_between_state_changes.sleep();
+        if (!rclcpp::ok()) {
+            return;
+        }
+        if (!gps_client->change_state(lifecycle_msgs::msg::Transition::TRANSITION_DEACTIVATE)) {
+            return;
+        }
+        if (!gps_client->get_state()) {
+            return;
+        }
+    }
+
+    // cleanup
+    {
+        time_between_state_changes.sleep();
+        if (!rclcpp::ok()) {
+            return;
+        }
+        if (!gps_client->change_state(lifecycle_msgs::msg::Transition::TRANSITION_CLEANUP)) {
+            return;
+        }
+        if (!gps_client->get_state()) {
+            return;
+        }
+    }
+
+    // and finally shutdown
+    {
+        time_between_state_changes.sleep();
```

Comment:
What's hapenning here? Why is is calling the shutdown transition?

Reply:
This was what the service client was going to be like, (individual files for each node),
I just forgot to delete this file.